src/main/cpp/_nix_based/jssc.cpp

```diff
@@ -580,17 +609,32 @@ static void awaitReadReady(JNIEnv*, jlong fd){
     fds[0].events = POLLIN;
     while(true){
         int result = poll(fds, 1, -1);
-        if(result < 0){
-            // man poll: On error, -1 is returned, and errno is set to indicate the error.
-            // TODO: Maybe a candidate to raise a java exception. But won't do
-            //       yet for backward compatibility.
-            continue;
+        if( result < 0 ){
+            err = errno;
+            jclass exClz = NULL;
+            switch( err ){
+            case EINVAL:
+                exClz = env->FindClass("java/lang/IllegalArgumentException");
+                if( exClz != NULL ) env->ThrowNew(exClz, "EINVAL poll()");
+                static_assert(EINVAL > 0, "EINVAL > 0");
+                return -EINVAL;
```

Comment:
Just code consistency with your other switch case.

```suggestion
                return -err;
```